### PR TITLE
Auto-create the Kafka topics if missing.

### LINKED ISSUE #1075: Kafka+SQL storage - auto create kafka topic if it is missing
When the application starts, the kafkasql storage should check for the existence of a Kafka topic of the appropriate name.  If no topic exists, we should create one with appropriate default settings.  This includes some (small?) number of partitions and log-compaction enabled.

### PR DESCRIPTION
**Pull request by EricWittmann**
Fixes #1075

### CODE REVIEW

File: utils/kafka/src/main/java/io/apicurio/registry/utils/kafka/KafkaUtil.java [hunk 1]
```diff
@@ -60,41 +63,61 @@
...
-                int minISR = Math.max(rf - 1, 1);
-                return new NewTopic(topicName, 1, (short) rf)
-                        .configs(Collections.singletonMap(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(minISR)));
+                log.info("Creating new Kafka topic: {}", topicName);
+                int replicationFactor = Math.min(3, nodes.size());
+                if (topicProperties.containsKey("replication.factor")) {
+                    replicationFactor = Integer.parseInt(topicProperties.get("replication.factor"));
+                }
+                int minimumInSyncReplicas = Math.max(replicationFactor - 1, 1);
+                topicProperties.putIfAbsent(TopicConfig.MIN_IN_SYNC_REPLICAS_CONFIG, String.valueOf(minimumInSyncReplicas));
```
Comment (alesj) [inline utils/kafka/src/main/java/io/apicurio/registry/utils/kafka/KafkaUtil.java:114]: Where is topicProperties used?

**Comment by alesj** [comment]
I created a new PR #1123 , which will break this one a bit.
But it should not be hard to sync it ...

File: storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlConfiguration.java [hunk 2]
```diff
@@ -25,11 +25,14 @@
 
     public String bootstrapServers();
     public String topic();
+    public Properties topicProperties();
+    public Boolean isTopicAutoCreate();
```
Comment (alesj) [inline storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlConfiguration.java:29]: Make it boolean ... ?

Comment (EricWittmann) [reply storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/KafkaSqlConfiguration.java:29]: Good idea.

File: utils/kafka/src/main/java/io/apicurio/registry/utils/kafka/KafkaUtil.java (hunk 1 above)
Comment (EricWittmann) [reply utils/kafka/src/main/java/io/apicurio/registry/utils/kafka/KafkaUtil.java:114]: Whoops!  It's supposed to be passed to `.configs()`.  Thanks for catching this.  Bonehead mistake.

**Review by carlesarnal — APPROVED** [review]

**Pull request merged**